### PR TITLE
[MISC] Add new issue template for bug reporting

### PR DESCRIPTION
**Pull request by Zheaoli**
Add new issue template for bug-reporting and feature-requesting

CleanShot 2022-01-27 at 17 10 26@2x

CleanShot 2022-01-27 at 17 10 48@2x

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/future_request.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Feature Request
+description: |
+  Report a bug encountered while using Taichi. 
+labels: 
+  - "feature request"
+body:
+  - type: textarea
+    id: problem
+    attributes:
+      label: Concisely describe the proposed feature
```
Comment (ailzhang) [inline .github/ISSUE_TEMPLATE/future_request.yaml:10]: and its use cases?

File: .github/ISSUE_TEMPLATE/future_request.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    validations:
+      required: true
+  - type: textarea
+    id: describe
+    attributes:
+      label: Describe the solution you'd like (if any)?
+      description: |
+        A clear and concise description of what you want to achieve and implement.
+    validations:
+      required: true
```
Comment (ailzhang) [inline .github/ISSUE_TEMPLATE/future_request.yaml]: I'm not sure what're the differences between this one and the box above ?

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        Please post a **minimal sample code** to reproduce the bug.
+        The developer team will put a higher priority on bugs that can be reproduced within 20 lines of code. If you want a prompt reply, please keep the sample code **short** and **representative**.
+      value: |
+        ```py
+        # sample code here
+        ```
+      render: markdown
+    validations:
+      required: true
+  - type: textarea
```
Comment (ailzhang) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:29]: To keep the template short, we could ask users to paste log in the "problem" section?

**Review by ailzhang — COMMENTED** [review]
Awesome! 
I made a few comments to make this template shorter - I'm afraid users might be scared away if we have too many required fields.

File: .github/ISSUE_TEMPLATE/future_request.yaml (hunk 2 above)
Comment (Zheaoli) [reply .github/ISSUE_TEMPLATE/future_request.yaml]: People can describe the impl detail here.!

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 3 above)
Comment (Zheaoli) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:29]: I think we can keep the detail here to help people give log as detail as they can

**Comment by Zheaoli** [comment]
>  I'm afraid users might be scared away if we have too many required fields.

I just think the developer will not delete unused information when they make a new issue lol

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 3 above)
Comment (ailzhang) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:29]: @Zheaoli Adding to my point above, I feel like it's hard to describe the problem without pasting the error msg/log? ;)

Comment (Zheaoli) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:29]: So I add some descriptions to remind people to add logs

CleanShot 2022-01-27 at 21 15 07@2x

**Comment by feisuzhu** [comment]
Closing ancient PR

**Pull request closed without merging**